### ISSUE TITLE
TileSet editor doesn't display selected tile properties

### ISSUE THREAD

**Issue by boruok** [godotengine/godot#88618]
### Tested versions

not reproducible:
- v3.6.beta4.official [93ac21cf8]
- v3.5.3.stable.official [6c814135b]

reproducible:
- v3.6.beta.custom_build [354404db6]
- v3.6.beta.custom_build [211c707e6]

### System information

Pop!_OS 22.04 LTS, GTX1060 545.29.06

### Issue description

tile properties not displayed

result
![Screenshot from 2024-02-21 12-33-59](https://github.com/godotengine/godot/assets/36783322/89356414-7d90-4a88-81a5-bdcdc45df110)
expected
![Screenshot from 2024-02-21 12-34-51](https://github.com/godotengine/godot/assets/36783322/ad1ac559-7a9a-4881-bf1f-65b93354cb4d)

bisect log
```console
git bisect start
# bad: [354404db6067997c1db358ba9c50668e93415b07] Merge pull request #76023 from Ansraer/3.x_orm
git bisect bad 354404db6067997c1db358ba9c50668e93415b07
# good: [93ac21cf8b0885dc595aa406699238ca4b2d580c] Merge pull request #87256 from akien-mga/3.x-cherrypicks
git bisect good 93ac21cf8b0885dc595aa406699238ca4b2d580c
# bad: [72ab4af489e12da4083943032f335e95a74c340c] Merge pull request #87739 from akien-mga/3.x-scons-web-workaround-emscripten-version-check
git bisect bad 72ab4af489e12da4083943032f335e95a74c340c
# bad: [1aa4c2fed11ce74b534893ce55177db4bc481eed] Merge pull request #86642 from Mickeon/3.x-doc-peeves-resourceful-uh-oh-stinky
git bisect bad 1aa4c2fed11ce74b534893ce55177db4bc481eed
# good: [4e741ec2ba36df2d3f9d54a729d45209b047b4a7] Merge pull request #64493 from Mickeon/3.x-editor-group
git bisect good 4e741ec2ba36df2d3f9d54a729d45209b047b4a7
# bad: [6f3c5e63f3bdc42ea3300900211a76469d03cdd0] Merge pull request #82584 from lawnjelly/lightcull_23
git bisect bad 6f3c5e63f3bdc42ea3300900211a76469d03cdd0
# bad: [24d1391af92d885072e3c9a1f80f1f114ae1bde3] Merge pull request #64931 from Mickeon/3.x-tileset-collision-group
git bisect bad 24d1391af92d885072e3c9a1f80f1f114ae1bde3
# bad: [211c707e67725f27ae995a923dd206cca9976315] [3.x] Add Selected Collision Group in TileSet Editor
git bisect bad 211c707e67725f27ae995a923dd206cca9976315
# first bad commit: [211c707e67725f27ae995a923dd206cca9976315] [3.x] Add Selected Collision Group in TileSet Editor
```

### Steps to reproduce

1. add tileset-resource
2. add tile
3. click tile
4. check tile properties

### Minimal reproduction project (MRP)

n/a

**Comment by lawnjelly** [comment]
#64931 is the PR causing the regression.

Immediately I can see that the if statement:
```
if (tileset_editor->edit_mode == TileSetEditor::EDITMODE_COLLISION && tileset_editor->edited_collision_shape.is_valid()) {
```
is now affecting display of the selected tile, which is presumably what is causing the regression.

**Comment by Mickeon** [comment]
I have already fixed this in a branch, I just would like to catch two birds with one stone if you know what I mean

**Comment by lawnjelly** [comment]
Usually it would be better to fix the regression in a PR asap, then add a second PR for the second bird. That way the 3.x branch is ideally kept bug free, and each change is reviewed independently. (Although I have no idea what your birds are :grin: ).

**Comment by Mickeon** [comment]
Alright, why not, I can do it now

**Comment by lawnjelly** [comment]
Fixed by #88828.
